_remake/lib/init-service-routes.js

```diff
@@ -0,0 +1,198 @@
+            expiresIn: config.jwt.duration
+          });
+          return res.status(200).json({ token }).end();
+        });
+      });
+  })
+
+  // endpoint for checking if subdomain is available (not already in the DB)
+  // validation callbacks: checkJWT, validSubdomain
+  // user must be authenticated to access it
+  app.get('/service/subdomain/check', checkJWT, validSubdomain, (req, res) => {
+    const { subdomain } = req.query;
+    connection.query('SELECT * FROM apps WHERE name = ?',
+      [subdomain],
+      (err, result, _) => {
+        if (err) return res.status(500).json(err).end();
+        if (result.length !== 0) return res.status(403).end();
+        return res.status(200).end();
+      });
+  });
+
+  // endpoint for registering subdomain (save in DB)
+  // validation callbacks: checkJWT, validSubdomain
+  // user must be authenticated to access it
+  app.post('/service/subdomain/register', checkJWT, validSubdomain, (req, res) => {
+    const { subdomain } = req.body;
+
+    connection.query('SELECT * FROM apps WHERE user_id = ?',
+      [req.user_id],
+      (err, results, fields) => {
+        if (err)
+          return res.status(500).json(err).end();
+        if (results.length >= global.config.limits.appPerUser)
+          return res.status(403).json({ message: `Reached ${global.config.limits.appPerUser} apps limit.` }).end();
+
+        connection.query('INSERT INTO apps (name, user_id, domain) VALUES (?, ?, ?)',
+          [subdomain, req.user_id, `${subdomain}.remakeapps.com`],
+          (err, results, fields) => {
+            if (err) {
+              if (err.code === "ER_DUP_ENTRY")
+                return res.status(400).json({ message: "An app with the same name already exists."}).end();
+              else return res.status(500).json(err).end();
+            }
+            return res.status(200).end();
+          });
+      });
+  });
+
+  // endpoint for deploying app
+  // upload app files if the user owns the app
+  // validation callbacks: checkJWT, validSubdomain, upload.single(...)
+  // user must be authenticated to access it
+  app.post('/service/deploy', checkJWT, upload.single('deployment'), validSubdomain, (req, res) => {
+    const { appName } = req.body;
+
+    connection.query('SELECT * FROM apps WHERE user_id = ? AND name = ?',
+      [req.user_id, appName],
+      (err, result, _) => {
+        if (err) return res.status(500).json(err).end();
+        if (result.length === 0) {
+          return res.status(403).json('Unauthorized to deploy').end();
+        }
+        extract(req.file.path, { dir: `${global.config.location.tmp}/${appName}` }, (err) => {
+          if (err) {
+            return res.status(500).json(err).end();
+          } else {
+            try {
+              shell.mkdir('-p', `${global.config.location.remake}/app/${appName}`);
+              shell.mkdir('-p', `${global.config.location.remake}/_remake-data/${appName}`);
```

Comment:
I don't think you need lines `179` and `180`, since you're using the `-p` flag on lines `181` and `182`

Reply:
👍